src/test/scala/com/amazon/deequ/analyzers/runners/AnalysisRunnerTests.scala

```diff
@@ -341,5 +341,23 @@ class AnalysisRunnerTests extends WordSpec with Matchers with SparkContextSpec w
         inputStream => assert(inputStream.read() > 0)
       }
     }
+
+    "should give error for duplicate analyzers" in withSparkSession { sparkSession =>
```

Comment:
I would add one more test case to make sure no exception raised when the same Analyzer added with various filtering options, e.g.
`val analyzers = Size() :: Size(Some("att1 = 0")) :: Size(Some("att2 > 0")) :: Nil`

This is a valid case, no exception should be thrown.

Reply:
Make sense, Added and it is passing

Reply:
@shashank-sharma 
You can use this pattern here to make the code more readable:
```
    noException shouldBe thrownBy {
      AnalysisRunner.onData(df)
        .addAnalyzers(analyzers)
        .useSparkSession(sparkSession)
        .run()
    }
```

Reply:
Yup, it is much better, Thanks I have made changes